app/src/main/java/org/mozilla/focus/utils/KeyboardUtil.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2015 Mike Penz All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Imported from MaterialDrawer:
+// https://github.com/mikepenz/MaterialDrawer/blob/fed00a199196efbc63296a081723e603647b5bc4/library/src/main/java/com/mikepenz/materialdrawer/util/KeyboardUtil.java
+package org.mozilla.focus.utils;
+
+
+import android.app.Activity;
+import android.graphics.Rect;
+import android.os.Build;
+import android.view.View;
+import android.view.ViewTreeObserver;
+import android.view.inputmethod.InputMethodManager;
+
+/**
+ * Created by mikepenz on 14.03.15.
+ * This class implements a hack to change the layout padding on bottom if the keyboard is shown
+ * to allow long lists with editTextViews
+ * Basic idea for this solution found here: http://stackoverflow.com/a/9108219/325479
+ */
+public class KeyboardUtil {
+    private View decorView;
+    private View contentView;
+
+    public KeyboardUtil(Activity act, View contentView) {
+        this.decorView = act.getWindow().getDecorView();
+        this.contentView = contentView;
+
+        //only required on newer android versions. it was working on API level 19
+        if (Build.VERSION.SDK_INT >= 19) {
+            decorView.getViewTreeObserver().addOnGlobalLayoutListener(onGlobalLayoutListener);
```

Comment:
We register a global layout listener here but we never unregister. Because of that we might keep references to the views and activity.